### PR TITLE
Phabricator task autoclose 

### PR DESCRIPTION
**Pull request by rebortg**
Add script and github action to close a task if it in the "Finished" column of all its projects.
the patched phabricator sdk is necessary for the "project.column.search" api endpoint

### CODE REVIEW

File: phabricator_tasks/phabricator_patched/__init__.py [hunk 1]
```diff
@@ -0,0 +1,404 @@
+"""
```
Comment (dmbaturin) [inline phabricator_tasks/phabricator_patched/__init__.py:1]: Did you write this module from scratch or it's a vendored library (or a part of a library)? Just wondering, we need to indicate code provenance in license headers like in all our repos.

File: phabricator_tasks/phabricator_patched/_compat.py [hunk 2]
```diff
@@ -0,0 +1,26 @@
...
+    from collections.abc import MutableMapping
+except ImportError:
+    from collections import MutableMapping
+
+try:
+    from urllib import urlencode
+except ImportError:
+    from urllib.parse import urlencode
+
+if PY3:
```
Comment (dmbaturin) [inline phabricator_tasks/phabricator_patched/_compat.py:15]: How can we end up with a worker host running Python2? We don't need compatibility with Python2 (which is dead), we need to make sure that the container image that runs this script is up to date. ;)

File: phabricator_tasks/phabricator_patched/tests/resources/certificate.txt [hunk 3]
```diff
@@ -0,0 +1 @@
+fdhcq3zsyijnm4h6gmh43zue5umsmng5t4dlwodvmiz4cnc6fl6fzrvjbfg2ftktrcddan7b3xtgmfge2afbrh4uwam6pfxpq5dbkhbl6mgaijdzpq5efw2ynlnjhoeqyh6dakl4yg346gbhabzkcxreu7hcjhw6vo6wwa7ky2sjdk742khlgsakwtme6sr2dfkhlxxkcqw3jngyrq5zj7m6m7hnscuzlzsviawnvg47pe7l4hxiexpbb5k456r
```
Comment (dmbaturin) [inline phabricator_tasks/phabricator_patched/tests/resources/certificate.txt:1]: What is this for? Don't we use Letsencrypt for the Phorge host, is there an issue with a missing CA?

File: phabricator_tasks/phabricator_patched/tests/resources/responses.json [hunk 4]
```diff
@@ -0,0 +1,5 @@
+{
+  "conduit.connect": "{\"result\":{\"connectionID\":1759,\"sessionKey\":\"lwvyv7f6hlzb2vawac6reix7ejvjty72svnir6zy\",\"userPHID\":\"PHID-USER-6ij4rnamb2gsfpdkgmny\"},\"error_code\":null,\"error_info\":null}",
```
Comment (dmbaturin) [inline phabricator_tasks/phabricator_patched/tests/resources/responses.json:2]: Just wondering, does the session key always remain the same?

**Review by dmbaturin — APPROVED** [review]

File: phabricator_tasks/phabricator_patched/__init__.py (hunk 1 above)
Comment (rebortg) [reply phabricator_tasks/phabricator_patched/__init__.py:1]: the code is from the this [repo](https://github.com/disqus/python-phabricator) the only thing i adding is the "project.column.search" api endpoint in the interface.json file.

**Comment by c-po** [comment]
I wonder if it's a good idea to add this PR into the terraform repository?

Maybe it makes more sense to generate a new phabricator-scripts repository?

**Comment by rebortg** [comment]
the date i create the PR the repo was empty, @sever-sever mention it, this is the wrong repository with a different purpose.

**Comment by dmbaturin** [comment]
Closing since we are going to use a different repository for it.

**Pull request closed without merging**